CMakeLists.txt

```diff
@@ -0,0 +1,1178 @@
+  libc-top-half/musl/src/complex/catan.c
+  libc-top-half/musl/src/complex/catanf.c
+  libc-top-half/musl/src/complex/catanh.c
+  libc-top-half/musl/src/complex/catanhf.c
+  libc-top-half/musl/src/complex/catanhl.c
+  libc-top-half/musl/src/complex/catanl.c
+  libc-top-half/musl/src/complex/ccos.c
+  libc-top-half/musl/src/complex/ccosf.c
+  libc-top-half/musl/src/complex/ccosh.c
+  libc-top-half/musl/src/complex/ccoshf.c
+  libc-top-half/musl/src/complex/ccoshl.c
+  libc-top-half/musl/src/complex/ccosl.c
+  libc-top-half/musl/src/complex/__cexp.c
+  libc-top-half/musl/src/complex/cexp.c
+  libc-top-half/musl/src/complex/__cexpf.c
+  libc-top-half/musl/src/complex/cexpf.c
+  libc-top-half/musl/src/complex/cexpl.c
+  libc-top-half/musl/src/complex/cimagl.c
+  libc-top-half/musl/src/complex/clog.c
+  libc-top-half/musl/src/complex/clogf.c
+  libc-top-half/musl/src/complex/clogl.c
+  libc-top-half/musl/src/complex/conj.c
+  libc-top-half/musl/src/complex/conjf.c
+  libc-top-half/musl/src/complex/conjl.c
+  libc-top-half/musl/src/complex/cpow.c
+  libc-top-half/musl/src/complex/cpowf.c
+  libc-top-half/musl/src/complex/cpowl.c
+  libc-top-half/musl/src/complex/cproj.c
+  libc-top-half/musl/src/complex/cprojf.c
+  libc-top-half/musl/src/complex/cprojl.c
+  libc-top-half/musl/src/complex/creall.c
+  libc-top-half/musl/src/complex/csin.c
+  libc-top-half/musl/src/complex/csinf.c
+  libc-top-half/musl/src/complex/csinh.c
+  libc-top-half/musl/src/complex/csinhf.c
+  libc-top-half/musl/src/complex/csinhl.c
+  libc-top-half/musl/src/complex/csinl.c
+  libc-top-half/musl/src/complex/csqrt.c
+  libc-top-half/musl/src/complex/csqrtf.c
+  libc-top-half/musl/src/complex/csqrtl.c
+  libc-top-half/musl/src/complex/ctan.c
+  libc-top-half/musl/src/complex/ctanf.c
+  libc-top-half/musl/src/complex/ctanh.c
+  libc-top-half/musl/src/complex/ctanhf.c
+  libc-top-half/musl/src/complex/ctanhl.c
+  libc-top-half/musl/src/complex/ctanl.c
+  libc-top-half/musl/src/crypt/crypt_blowfish.c
+  libc-top-half/musl/src/crypt/crypt.c
+  libc-top-half/musl/src/crypt/crypt_des.c
+  libc-top-half/musl/src/crypt/crypt_md5.c
+  libc-top-half/musl/src/crypt/crypt_r.c
+  libc-top-half/musl/src/crypt/crypt_sha256.c
+  libc-top-half/musl/src/crypt/crypt_sha512.c
+  libc-top-half/musl/src/crypt/encrypt.c
+  libc-top-half/sources/arc4random.c)
+target_include_directories(libc-top-half PRIVATE
+  libc-top-half/musl/src/include
+  libc-top-half/musl/src/internal
+  libc-top-half/musl/arch/wasm32
+  libc-top-half/musl/arch/generic
+  libc-top-half/headers/private)
+target_compile_options(libc-top-half PRIVATE
+  -Wno-parentheses
+  -Wno-shift-op-parentheses
+  -Wno-bitwise-op-parentheses
+  -Wno-logical-op-parentheses
+  -Wno-string-plus-int
+  -Wno-dangling-else
+  -Wno-unknown-pragmas)
+add_dependencies(libc-top-half sysroot-headers)
+
+target_link_libraries(c PRIVATE
+  libc-bottom-half
+  libc-top-half
+  dlmalloc)
+
+foreach(stub m rt pthread crypt util xnet resolve dl)
```

Comment:
Could you preserve the original comment "Create empty placeholder libraries" here?